nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/StandardProcessorNode.java

```diff
@@ -1404,9 +1389,9 @@ private void invokeTaskAsCancelableFuture(ScheduledExecutorService taskScheduler
                     + "have been written to ignore interrupts which may result in runaway thread which could lead to more issues "
                     + "eventually requiring NiFi to be restarted. This is usually a bug in the target Processor '"
                     + this.processor + "' that needs to be documented, reported and eventually fixed.");
+            throw new RuntimeException("Timed out while executing one of processor's lifecycle tasks (OnScheduled or OnUnscheduled).", e);
         } catch (ExecutionException e){
-            throw new RuntimeException(
-                    "Failed while executing one of processor's lifecycle tasks (OnScheduled or OnUnscheduled).", e);
+            throw new RuntimeException("Failed while executing one of processor's lifecycle tasks (OnScheduled or OnUnscheduled).", e);
```

Comment:
Can we pass in the name of the lifecycle event here or something? I would avoid indicating "OnScheduled or OnUnscheduled" -- we should know when we generate this message which one it was.